modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -742,7 +742,6 @@ OPENCV_HAL_IMPL_MATH_FUNC(v_sqrt, std::sqrt, _Tp)
  * @note The precision of the calculation depends on the implementation and the data type of the input vector.
  */
 OPENCV_HAL_IMPL_MATH_FUNC(v_exp, std::exp, _Tp)
-#define OPENCV_HAL_MATH_HAVE_EXP 1
```

Comment:
It is expected that tests like `hal_intrin*CPP_EMULATOR` call the code from `intrin_cpp.hpp`. But they don't.

To debug modify `intrin_cpp.hpp`:

```.diff
 #define OPENCV_HAL_IMPL_MATH_FUNC(func, cfunc, _Tp2) \
 template<typename _Tp, int n> inline v_reg<_Tp2, n> func(const v_reg<_Tp, n>& a) \
 { \
+    printf(#func " is called\n"); \
     v_reg<_Tp2, n> c; \
     for( int i = 0; i < n; i++ ) \
         c.s[i] = cfunc(a.s[i]); \

```

Run tests: `./bin/opencv_test_core --gtest_filter=*intrin*EMULATOR*`

```
[ RUN      ] hal_intrin128.float32x4_CPP_EMULATOR
SIMD128: void opencv_test::hal::intrin128::opt_EMULATOR_CPP::test_hal_intrin_float32()
v_abs is called
v_sqrt is called
v_abs is called
<<<<<<< missing v_exp call through generic C++ exp() >>>>>>>>>
v_abs is called
v_abs is called
v_abs is called
...
```

(both build configurations: with/without -DOPENCV_EXTRA_FLAGS="-DCV_FORCE_SIMD128_CPP=1")


---

BTW, without this PR the correct version is called (in "default" build):

```
[ RUN      ] hal_intrin128.float32x4_CPP_EMULATOR
SIMD128: void opencv_test::hal::intrin128::opt_EMULATOR_CPP::test_hal_intrin_float32()
v_abs is called
v_sqrt is called
v_abs is called
v_exp is called
v_exp is called
v_exp is called
v_exp is called
v_exp is called
...
v_log is called
...

```

Reply:
This happens may because we introduce `v_erf` function, but the test call v_exp with the implementation in `intrin_math.hpp`.  I think I should add a line after erf defination: `#define OPENCV_HAL_MATH_HAVE_ERF 1`

```c++
/**
 * @brief Error function.
 *
 * @note Support FP32 precision for now.
 */
OPENCV_HAL_IMPL_MATH_FUNC(v_erf, std::erf, _Tp)
```